src/libmeasurement_kit/ooni/meek_fronted_requests.cpp

```diff
@@ -0,0 +1,93 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include "../common/utils.hpp"
+#include "../ooni/constants.hpp"
+#include <measurement_kit/ooni.hpp>
+
+namespace mk {
+namespace ooni {
+
+using namespace mk::report;
+
+void meek_fronted_requests(std::string input, Settings options,
+                   Callback<Var<report::Entry>> callback,
+                   Var<Reactor> reactor, Var<Logger> logger) {
+    Var<Entry> entry(new Entry);
+
+    std::string expected_body, outer_host, inner_host;
+
+    if (options["expected_body"].empty()) {
+        expected_body = constants::MEEK_SERVER_RESPONSE;
+    } else {
+        expected_body = options["expected_body"];
+    }
+
+    if (!options["outer_host"].empty() &&
+        !options["inner_host"].empty()) {
+        outer_host = options["outer_host"];
+        inner_host = options["inner_host"];
+    } else {
+        std::list<std::string> outer_inner = split(input, ":");
+        if (outer_inner.size() != 2) {
+            logger->debug("Couldn't split input: %s", input.c_str());
+            callback(entry);
+            return;
+        }
+        // XXX: We should make sure that we remove leading and trailing whitespaces
+        outer_host = outer_inner.front();
+        inner_host = outer_inner.back();
+    }
+
+    // url parsing methods require a schema
+    outer_host.insert(0, "https://");
+    inner_host.insert(0, "https://");
```

Comment:
If what you need to parse is an endpoint (i.e. `<ADDRESS_OR_HOST>:<PORT>`, [we have code for doing that](https://github.com/measurement-kit/measurement-kit/blob/master/include/measurement_kit/net/utils.hpp#L12).